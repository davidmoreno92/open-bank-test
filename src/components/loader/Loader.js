import React, { Component } from 'react'
import CircularProgress from '@material-ui/core/CircularProgress'

import './Loader.scss';

export default class Loader extends Component {
    
    render() {
        return (
            <div class="loader">
                <CircularProgress />
            </div>
        )
    }
}