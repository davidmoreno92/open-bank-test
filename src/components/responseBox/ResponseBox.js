import React from 'react';

import './ResponseBox.scss';

function ResponseBox(props) {
    const { imgPath, title, message, className } = props;

    return (
        <div className={`row ${className ? className : ""}`} p={1}>
            {imgPath ?
                <div>
                    <img src={imgPath} alt="fake alt" />
                </div>
                : ''
            }
            <div>
                {title ?
                    <h1>
                        {title}
                    </h1>
                    : ''
                }
                {message ?
                    <p>
                        {message}
                    </p>
                    : ''
                }
            </div>
        </div>
    )
}


export default ResponseBox;