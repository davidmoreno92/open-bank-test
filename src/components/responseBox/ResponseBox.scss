@import '../../assets/scss/variables';


.buttons-box {
    @include small {
        margin: 0 auto;
        padding: 2rem 6rem;
        border-top: 1px solid #ccc;
        margin-top: 2rem;
    }

    display: flex;
    justify-content: space-between;
    align-items: center;

    button {
        font-weight: bold;
        background-color: $blue-dark-color;
        color: #fff;
        padding: 0.5rem 1.5rem;
        border: 0;
        border-radius: 0.2rem;
        text-transform: none;

        &:hover {
            //To override material hover
            background-color: $blue-dark-color
        }

        &.transparent {
            color: $blue-dark-color;
            background-color: transparent;
            border: 0;

            &.pink {
                color: $pink-color;

                .arrow {
                    border-color: $pink-color;
                }
            }
        }

        .arrow {
            border: solid #fff;
            border-width: 0 3px 3px 0;
            display: inline-block;
            padding: 3px;
            margin: 0.5rem;
        }

        .right {
            transform: rotate(-45deg);
            -webkit-transform: rotate(-45deg);
        }
    }

    .right {
        margin-left: auto;
    }

}
