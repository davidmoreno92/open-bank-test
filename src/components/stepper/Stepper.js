import React, { Component } from "react"

import './Stepper.scss';

export default class Stepper extends Component {
    render() {
        const { steps, step } = this.props;

        return (
            <div className="stepper-container">
                {
                    steps.map((element, index) => {
                        const currentStep = step;
                        const lastStep = step.length - 1;

                        const isCurrentStep = currentStep === index;
                        const isLastStep = currentStep === lastStep;

                        return (
                            <span>
                                <div className={`stepper ${!isLastStep && isCurrentStep ? "active" : isLastStep === 2 || currentStep > index ? "done" : ''}`}>
                                    {isCurrentStep && <div className="triangle"></div>}
                                    {currentStep > index || isLastStep ? <span className="number">&#10004;</span> : <span className="number">{index + 1}</span>}
                                </div>
                                {index !== lastStep && <div className={`separator ${currentStep > index || isLastStep ? "done" : ''}`} />}
                            </span>
                        )
                    })
                }
            </div>
        )
    }
}