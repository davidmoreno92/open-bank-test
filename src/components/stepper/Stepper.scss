@import '../../assets/scss/variables';

.triangle {
    border-style: solid;
    border-width: 0 15px 15px 15px;
    border-color: transparent transparent #ffffff transparent;
    position: absolute;
    bottom: 0;
    filter: drop-shadow(0px -2px 1px rgba(0, 0, 0, .3));
    margin-left: 0.1rem;
}

.stepper-container {
    position: relative;
    padding: 1.5rem 0;
    background-color: #ccd5da;
    -moz-box-shadow: inset 0 -10px 10px -10px #545454;
    -webkit-box-shadow: inset 0 -10px 10px -10px #545454;
    box-shadow: inset 0 -10px 5px -10px #545454;
    text-align: center;

    .separator {
        display: inline-block;
        width: 48px;
        height: 3px;
        background-color: $bg-color;
        vertical-align: middle;

        &.done {
            background-color: $pink-color;
        }
    }

    .stepper {
        display: inline-flex;
        align-items: center;

        &.active {
            .number {
                background-color: $blue-dark-color;
                padding: 0.5rem;
            }
        }

        &.done {
            .number {
                background-color: $pink-color;
            }
        }

        .number {
            background-color: $blue-color;
            display: inline-flex;
            align-items: center;
            justify-content: center;
            height: 20px;
            width: 20px;
            font-size: 10px;
            border-radius: 50%;
            color: white;
            font-size: 1rem;
            padding: 0.3rem;
        }
    }


}