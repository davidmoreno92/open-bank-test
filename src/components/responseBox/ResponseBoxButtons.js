import React, { Component } from 'react';
import { connect } from 'react-redux'

import Button from '@material-ui/core/Button'
import Box from '@material-ui/core/Box'
import { resetForm } from '../../actions/FormAction'
import './ResponseBox.scss';

class ResponseBoxButtons extends Component {
    render() {
        const { message } = this.props;
        return (
            <Box className="buttons-box">
                <Box className="right">
                    <Button onClick={() => this.props.resetForm()} className="transparent pink">
                        {message ? message : 'Volver a Password Manager'}
                        <i className="arrow right" />
                    </Button>
                </Box>
            </Box>
        )
    }
}

const mapStateToProps = (state) => ({

})

const mapDispatchToProps = (dispatch) => ({
    resetForm: () => dispatch(resetForm())
})



export default connect(mapStateToProps, mapDispatchToProps)(ResponseBoxButtons)