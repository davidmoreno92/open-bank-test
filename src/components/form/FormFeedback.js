import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withTranslation } from 'react-i18next'

import { sendFormData } from '../../actions/FormAction'
import ResponseBox from '../responseBox/ResponseBox'
import ResponseBoxButtons from '../responseBox/ResponseBoxButtons'
import Loader from '../loader/Loader'
import './Form.scss';

class FormFeedback extends Component {

    async componentDidMount() {
        await this.props.sendFormData({ password: this.props.password, passwordRepeat: this.props.passwordRepeat, clue: this.props.clue });
    }

    render() {
        const { isSending, responseData, responseOK, t } = this.props;
        const isOk = responseData.status === responseOK;

        const imgPath = isOk ? "/images/success-icon.jpg" : "/images/error-icon.jpg";
        const title = isOk ? t('form.feedback.success.title') : t('form.feedback.failed.title');
        const message = isOk ? t('form.feedback.success.message'): t('form.feedback.failed.message');
        const buttonMessage = isOk ? t('form.feedback.success.button-text') : t('form.feedback.failed.button-text');
        const buttonPath = isOk ? "/" : "";

        return (
            <div>
                {!isSending && responseData ?
                    <>
                        <div className="stepFeedback">
                            <ResponseBox className="notification-row" imgPath={imgPath} title={title} message={message} />
                        </div>
                        <ResponseBoxButtons onClick={this.clickHandler} message={buttonMessage} path={buttonPath} />
                    </>
                    :
                    <Loader />
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    isSending: state.FormReducer.isSending,
    responseData: state.FormReducer.responseData,
    password: state.FormReducer.password,
    passwordRepeat: state.FormReducer.passwordRepeat,
    clue: state.FormReducer.clue,
    responseOK: state.FormReducer.responseOK
})

const mapDispatchToProps = (dispatch) => ({
    sendFormData: (formData) => sendFormData(dispatch, formData)
})

export default connect(mapStateToProps, mapDispatchToProps)(withTranslation()(FormFeedback))