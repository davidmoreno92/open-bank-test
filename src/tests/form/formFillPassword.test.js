import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Loader from '../../components/loader/Loader';
import Stepper from '../../components/stepper/Stepper';
import ResponseBox from '../../components/responseBox/ResponseBox';
import { RESPONSE_OK } from '../../services/api';
import ResponseBoxButtons from '../../components/responseBox/ResponseBoxButtons';
import FormFeedback from '../../components/form/FormFeedback';

const responseIsOk = RESPONSE_OK; //RESPONSE_KO
const mockStore = configureMockStore();
const store = mockStore({});

const props = { responseData: { status: 200 } };
describe("Testpage Component", () => {
  console.log(props);
  it("should render without throwing an error and find success text", () => {
    expect(
      shallow(
        <Provider store={store}>
          <FormFeedback props={props} />
        </Provider>
      ).exists(<h1>¡Tu Password Manager ya está creado!</h1>)
    ).toBe(true);
  });
});

describe("Loader Component", () => {
  test("Matches the snapshot", () => {
    const loader = renderer.create(<Loader />);
    expect(loader.toJSON()).toMatchSnapshot();
  });
});

describe("Stepper Component", () => {
  test("Matches the snapshot", () => {
    const step = 1;
    const steps = ["FormProduct", "FormPassword", "FormFeedback"];
    const loader = renderer.create(<Stepper step={step} steps={steps} />);
    expect(loader.toJSON()).toMatchSnapshot();
  });
});

describe("Response Box Component", () => {
  test("Matches the snapshot", () => {
    const imgPath = responseIsOk ? "/images/success-icon.jpg" : "/images/error-icon.jpg";
    const title = responseIsOk ? 'form.feedback.success.title' : 'form.feedback.failed.title';
    const message = responseIsOk ? 'form.feedback.success.message' : 'form.feedback.failed.message';

    const loader = renderer.create(<ResponseBox imgPath={imgPath} title={title} message={message} />);
    expect(loader.toJSON()).toMatchSnapshot();
  });
});

describe("Response Box Buttons Component", () => {
  test("Matches the snapshot", () => {
    const buttonMessage = responseIsOk ? 'form.feedback.success.button-text' : 'form.feedback.failed.button-text';
    const buttonPath = responseIsOk ? "/" : "";

    const loader = renderer.create(<ResponseBoxButtons message={buttonMessage} path={buttonPath} />);
    expect(loader.toJSON()).toMatchSnapshot();
  });
});