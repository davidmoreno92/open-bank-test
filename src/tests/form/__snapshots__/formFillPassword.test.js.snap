// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Matches the snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`Response Box Buttons Component Matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-1 buttons-box"
>
  <div
    className="MuiBox-root MuiBox-root-2 right"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text transparent pink"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        form.feedback.success.button-text
        <i
          className="arrow right"
        />
      </span>
    </button>
  </div>
</div>
`;

exports[`Response Box Component Matches the snapshot 1`] = `
<div
  className="row "
  p={1}
>
  <div>
    <img
      alt="fake alt"
      src="/images/success-icon.jpg"
    />
  </div>
  <div>
    <h1>
      form.feedback.success.title
    </h1>
    <p>
      form.feedback.success.message
    </p>
  </div>
</div>
`;

exports[`Stepper Component Matches the snapshot 1`] = `
<div
  className="stepper-container"
>
  <span>
    <div
      className="stepper done"
    >
      <span
        className="number"
      >
        âœ”
      </span>
    </div>
    <div
      className="separator done"
    />
  </span>
  <span>
    <div
      className="stepper active"
    >
      <div
        className="triangle"
      />
      <span
        className="number"
      >
        2
      </span>
    </div>
    <div
      className="separator "
    />
  </span>
  <span>
    <div
      className="stepper "
    >
      <span
        className="number"
      >
        3
      </span>
    </div>
  </span>
</div>
`;
